src/integrations/prefect-dbt/prefect_dbt/core/settings.py

```diff
@@ -2,16 +2,99 @@
 A class for configuring or automatically discovering settings to be used with PrefectDbtRunner.
 """
 
+import contextlib
+import tempfile
 from pathlib import Path
+from typing import Any, Generator
 
+import yaml
 from dbt_common.events.base_types import EventLevel
 from pydantic import Field
 from pydantic_settings import BaseSettings, SettingsConfigDict
 
+from prefect.settings import get_current_settings
+from prefect.utilities.asyncutils import run_coro_as_sync
+from prefect.utilities.templating import (
+    resolve_block_document_references,
+    resolve_variables,
+)
+from prefect_dbt.utilities import find_profiles_dir, replace_with_env_var_call
+
 
 class PrefectDbtSettings(BaseSettings):
+    """
+    dbt settings that directly affect the PrefectDbtRunner.
+    These settings will be collected automatically from their corresponding 'DBT_'-prefixed environment variables.
+    If a setting is not set in the environment or in the fields of this class, the default value will be used.
+
+    All other dbt settings should be used as normal, e.g. in the dbt_project.yml file, env vars, or kwargs to `invoke()`.
+    """
+
     model_config = SettingsConfigDict(env_prefix="DBT_")
 
-    profiles_dir: Path = Field(default=Path.home() / ".dbt")
-    project_dir: Path = Field(default_factory=Path.cwd)
-    log_level: EventLevel = Field(default=EventLevel.INFO)
+    profiles_dir: Path = Field(
+        default_factory=find_profiles_dir,
+        description="The directory containing the dbt profiles.yml file.",
+    )
+    project_dir: Path = Field(
+        default_factory=Path.cwd,
+        description="The directory containing the dbt project.",
+    )
+    log_level: EventLevel = Field(
+        default=EventLevel(get_current_settings().logging.level.lower()),
```

Comment:
i think this should be a default_factory so you don't get the setting value at import time, since you can modify them at runtime